sotodlib/site_pipeline/make_source_flags.py

```diff
@@ -0,0 +1,113 @@
+from argparse import ArgumentParser
+import logging
+import numpy as np
+import os
+import sys
+import yaml
+
+import sotodlib
+from sotodlib import core, coords, site_pipeline
+
+logger = logging.getLogger(__name__)
+
+def get_parser():
+    parser = ArgumentParser()
+    parser.add_argument('obs_id',help=
+                        "Observation for which to generate flags.")
+    parser.add_argument('-c', '--config-file', help=
+                        "Configuration file.")
+    parser.add_argument('-v', '--verbose', action='count',
+                        default=0, help="Pass multiple times to increase.")
+    return parser
+
+def get_config(args):
+    cfg = yaml.safe_load(open(args.config_file, 'r'))
+    for k in ['obs_id', 'verbose']:
+        cfg[k] = getattr(args, k)
+    return cfg
+
+def main(args=None):
+    if args is None:
+        args = sys.argv[1:]
+    parser = get_parser()
+    config = get_config(parser.parse_args(args))
+
+    if config['verbose'] >= 1:
+        logger.setLevel('INFO')
+    if config['verbose'] >= 2:
+        sotodlib.logger.setLevel('INFO')
+    if config['verbose'] >= 3:
+        sotodlib.logger.setLevel('DEBUG')
+
+    ctx = core.Context(config['context_file'])
+
+    group_by = config['subobs'].get('use', 'detset')
+    if group_by == 'detset':
+        groups = ctx.obsfiledb.get_detsets(config['obs_id'])
+    elif group_by.startswith('dets:'):
+        group_by = group_by.split(':',1)[1]
+        groups = ctx.detdb.props(props=group_by).distinct()[group_by]
+    else:
+        raise ValueError("Can't group by '{group_by}'")
+
+    if os.path.exists(config['archive']['index']):
+        logger.info(f'Mapping {config["archive"]["index"]} for the archive index.')
+        db = core.metadata.ManifestDb(config['archive']['index'])
+    else:
+        logger.info(f'Creating {config["archive"]["index"]} for the archive index.')
+        scheme = core.metadata.ManifestScheme()
+        scheme.add_exact_match('obs:obs_id')
+        if group_by != 'detset':
+            scheme.add_exact_match('dets:' + group_by)
+        scheme.add_data_field('dataset')
+        db = core.metadata.ManifestDb(config['archive']['index'], scheme=scheme)
+
+    for group in groups:
+        logger.info(f'Loading {config["obs_id"]}:{group_by}={group}')
+        if group_by == 'detset':
+            # Load data.  We don't need all the signal, btw.
+            tod = ctx.get_obs(config['obs_id'], detsets=[group])
```

Comment:
Yes, totally.  It's confusing because one would want to have the .dets axis created properly but have .signal be None or not present.  Probably None.  Let's plan to do that.